app/src/main/java/com/hedvig/app/feature/embark/passages/multiaction/add/AddComponentBottomSheet.kt

```diff
@@ -0,0 +1,135 @@
+package com.hedvig.app.feature.embark.passages.multiaction.add
+
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import androidx.core.os.bundleOf
+import androidx.fragment.app.setFragmentResult
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import com.hedvig.app.R
+import com.hedvig.app.databinding.DialogAddBuildingBinding
+import com.hedvig.app.feature.embark.passages.multiaction.MultiAction
+import com.hedvig.app.feature.embark.passages.multiaction.MultiActionParams
+import com.hedvig.app.util.extensions.hideKeyboard
+import com.hedvig.app.util.extensions.onImeAction
+import com.hedvig.app.util.extensions.viewBinding
+import kotlinx.android.synthetic.main.dialog_add_building.numberLayout
+import kotlinx.android.synthetic.main.picker_button.continueButton
+import org.koin.android.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
+
+class AddComponentBottomSheet : BottomSheetDialogFragment() {
```

Comment:
I'm not sure that this implementation will work on spec: `MultiAction` should dynamically create a form based on `EmbarkMultiActionData.components`, this seems to be strictly tied to the one implementation we have of `MultiAction` today, but we need to implement this to work generically. 😕

Reply:
Oh, Yeah I was a bit confused about the data structure for `EmbarkMultiActionComponent`. So we can get a list with components each containing a number, dropdown, or switch-action? I basically want to create multiple `dropdownLayout`, `numberLayout` and `componentSwitch` for each component?